src/main/webapp/script.js

```diff
@@ -260,3 +280,137 @@ function checkPercentage(headerColumn) {
+    const countyAndStateArray = county[0].split(',');
+    // ^^ ["Contra Costa County", "California"]
+    const countyArray = countyAndStateArray[0].split(' ');
+    // ^^ ["Contra", "Costa", "County"]
+    let countyString = '';
+    let i;
+    // Get all strings except for the last one
+    for (i = 0; i < countyArray.length - 1; i++) {
+      countyString += countyArray[i];
+      if (i !== countyArray.length - 2) {
+        countyString += ' ';
+      }
+    }
+    // Map the population to the county
+    countyToPopMap[countyString] = county[1];
+    populationsList.push(parseInt(county[1]));
+    });
+  const minAndMax = getMinAndMaxPopulation(populationsList);
+  return {map: countyToPopMap,
+    minValue: minAndMax.min, maxValue: minAndMax.max};
+}
+
+// Returns an object with the min and max population of the
+// populations returned by the census API.
+function getMinAndMaxPopulation(populationArray) {
+  let max = populationArray[0];
+  let min = 0;
+  let i;
+  for (i = 1; i < populationArray.length; i++) {
+    if (populationArray[i] > max) {
+      max = populationArray[i];
+    } else if (populationArray[i] < min) {
+      min = populationArray[i];
+    }
+  }
+  return {max: max, min: min};
+}
+
+// Takes in mapsData object which has a data structure that maps
+// counties to populations, a max population, and a min population.
+// Initializes the geoJson and adds multiple event listeners.
+function displayCountyGeoJson(mapsData, stateName) {
+  const map = new google.maps.Map(document.getElementById('map'), {
+    zoom: stateInfo[stateName].zoomLevel,
+    center: {lat: stateInfo[stateName].lat, lng: stateInfo[stateName].lng},
+  });
+
+  const countyToPopMap = mapsData.map;
+  const maxPopulation = mapsData.maxValue;
+  const minPopulation = mapsData.minValue;
+  const f = chroma.scale(['white', 'blue']).domain([minPopulation,
+    maxPopulation]);
+  const geoData = getGeoData(stateName, true);
+
+  map.data.addGeoJson(geoData);
+  map.data.forEach(function(feature) {
+    map.data.setStyle((feature) => {
+      return {
+        fillColor: f(countyToPopMap[feature.j.name]).toString(),
+      };
+    });
+  });
+
+  const openInfoWindows = [];
+  map.data.addListener('mouseover', function(event) {
+    map.data.overrideStyle(event.feature, {
+      fillColor: '#00ffff',
+    });
+    const contentString = '<p>' + event.feature.j.name +
+    '<p>Population: ' + countyToPopMap[event.feature.j.name];
+    const infoWindow = new google.maps.InfoWindow({
+      content: contentString,
+      maxWidth: 100,
+    });
+    infoWindow.setPosition(event.latLng);
+    infoWindow.open(map);
+    openInfoWindows.push(infoWindow);
+  });
+  map.data.addListener('mouseout', function(event) {
+    map.data.revertStyle();
+    let i;
+    for (i = 0; i < openInfoWindows.length; i++) {
+      openInfoWindows[i].close();
+    }
+  });
+}
+
+// Functions to toggle between amcharts and maps.
+function showGeoJson() {
```

Comment:
These two functions are very similar, maybe they could just be one function and take in some parameters to determine which is made visible.

Reply:
I think you can pass in a parameter and then decide which to display and which to remove.

Reply:
Fixed to a toggle() function that takes in parameters.